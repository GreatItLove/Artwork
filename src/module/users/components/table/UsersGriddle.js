import React, { Component } from 'react';
import Griddle, { RowDefinition, ColumnDefinition } from 'griddle-react';
import PropTypes from 'prop-types';
import { RefreshIndicator } from 'material-ui';

import CustomActionCell from './CustomActionCell';

import CustomActiveCell from '../../../common/components/table/CustomActiveCell';
import CustomLastLoggedInCell from '../../../common/components/table/CustomLastLoggedInCell';
import CustomNextButton from '../../../common/components/table/CustomNextButton';
import CustomPreviousButton from '../../../common/components/table/CustomPreviousButton';
import CustomShowPerPage from '../../../common/components/table/CustomShowPerPage';
import CustomFilter from '../../../common/components/filter/CustomFilter';

import DeleteUserDialog from '../../containers/DeleteUserDialog';
import SuspendUserDialog from '../../containers/SuspendUserDialog';
import AddUserDialog from '../../containers/AddUserDialog';
import EditUserDialog from '../../containers/EditUserDialog';
import ResetPasswordDialog from '../../containers/ResetPasswordDialog';
import DownloadUsersDialog from '../../containers/DownloadUsersDialog';
import {
  fetchUsersRequest,
  updateSearchFilter,
  switchUsersFilter,
} from '../../usersActions';

const propTypes = {
  list: PropTypes.array.isRequired,
  appDispatch: PropTypes.func.isRequired,
};

const tableStyleConfig = {
  classNames: {
    Table: 'table table-hover',
  },
};

/**
 * @class UsersGriddle
 * @extends React.Component
 * @description Render component
 */
class UsersGriddle extends Component {
  /**
   * @description render
   * @return {Object} JSX HTML Content
   */
  _onFilter = filterText => {
    const { appDispatch, pagination } = this.props;

    appDispatch(updateSearchFilter(filterText));
    appDispatch(
      fetchUsersRequest({ limit: pagination.recordsPerPage, ...filterText }),
    );
  };
  _onNext = () => {
    const { appDispatch, pagination } = this.props;
    appDispatch(
      fetchUsersRequest({
        limit: pagination.recordsPerPage,
        offset: pagination.currentPage * pagination.recordsPerPage,
        ...this.props.searchFilter,
      }),
    );
  };

  _onPrevious = () => {
    const { appDispatch, pagination } = this.props;
    appDispatch(
      fetchUsersRequest({
        limit: pagination.recordsPerPage,
        offset: (pagination.currentPage - 2) * pagination.recordsPerPage,
        ...this.props.searchFilter,
      }),
    );
  };

  _onGetPage = pageNumber => {
    const { appDispatch, pagination } = this.props;
    appDispatch(
      fetchUsersRequest({
        limit: pagination.recordsPerPage,
        offset: (pageNumber - 1) * pagination.recordsPerPage,
        ...this.props.searchFilter,
      }),
    );
  };

  render() {
    const { list, appDispatch, pagination, t } = this.props;
    let filterData = {
      name: 'Users',
      radio: [
        {
          name: 'active',
          fields: [{ name: 'Yes', value: true }, { name: 'No', value: false }],
        },
        {
          name: 'userType',
          fields: [
            { name: 'SA', value: 'SA' },
            { name: 'Staff', value: 'Staff' },
            { name: 'Gallery', value: 'Gallery' },
            { name: 'Admin', value: 'Admin' },
            { name: 'Webmaster', value: 'Webmaster' },
          ],
        },
        {
          name: 'userLevel',
          fields: [
            { name: '0', value: '0' },
            { name: '1', value: '1' },
            { name: '2', value: '2' },
            { name: '3', value: '3' },
            { name: '4', value: '4' },
          ],
        },
      ],
    };
    return (
      <div>
        <Griddle
          data={list}
          styleConfig={tableStyleConfig}
          pageProperties={{
            currentPage: pagination.currentPage,
            pageSize: pagination.recordsPerPage,
            recordCount: pagination.totalRecordCount,
          }}
          events={{
            onFilter: this._onFilter,
            onNext: this._onNext,
            onPrevious: this._onPrevious,
            onGetPage: this._onGetPage,
          }}
          components={{
            SettingsWrapper: () => null,
            Filter: filter => {
              return (
                <CustomFilter
                  t={t}
                  showingFilter={this.props.showingFilter}
                  searchFilter={this.props.searchFilter}
                  recordCount={this.props.pagination.totalRecordCount}
                  appDispatch={appDispatch}
                  switchFilter={switchUsersFilter}
                  filterData={filterData}
                  setFilter={filter.setFilter}
                  className={filter.className}
                  style={filter.style}
                />
              );
            },
            PageDropdown: () => null,
            NextButton: CustomNextButton,
            PreviousButton: CustomPreviousButton,
            Table: ({
              className,
              style,
              visibleRows,
              TableHeading,
              TableBody,
              NoResults,
            }) => {
              return this.props.fetching ? (
                <div
                  className={className}
                  style={{
                    position: 'absolute',
                    bottom: '120px',
                    left: '48%',
                    ...style,
                  }}>
                  <RefreshIndicator left={0} top={0} status="loading" />
                </div>
              ) : visibleRows ? (
                <table className={className} style={style}>
                  <TableHeading />
                  <TableBody />
                </table>
              ) : (
                    <NoResults />
                  );
            },
          }}>
          <RowDefinition>
            <ColumnDefinition id="fullName" title={t("columns.name")} />
            <ColumnDefinition id="email" title={t("columns.email")} />
            <ColumnDefinition
              id="lastLogin"
              title={t("columns.lastLoggedin")}
              customComponent={CustomLastLoggedInCell}
            />
            <ColumnDefinition id="userType" title={t("columns.type")} />
            <ColumnDefinition id="userLevel" title={t("columns.level")} />
            <ColumnDefinition
              id="active"
              title={t("columns.active")}
              customComponent={CustomActiveCell}
            />
            <ColumnDefinition
              id="userId"
              title=" "
              customComponent={CustomActionCell}
              extraData={{ appDispatch }}
            />
          </RowDefinition>
        </Griddle>
        <CustomShowPerPage fetchRequest={fetchUsersRequest} {...this.props} />
        <DeleteUserDialog />
        <SuspendUserDialog />
        <AddUserDialog />
        <EditUserDialog />
        <ResetPasswordDialog />
        <DownloadUsersDialog />
      </div>
    );
  }
}

UsersGriddle.propTypes = propTypes;

export default UsersGriddle;
