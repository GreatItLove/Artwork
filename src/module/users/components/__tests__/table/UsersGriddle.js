import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import UsersGriddle from '../../table/UsersGriddle';
const userList = [
  {
    active: true,
    email: 'demo@artworkmanager.com',
    fullName: 'Andrew Hershey',
    lastLogin: '1505204346',
    userId: 6733,
    userLevel: '4         ',
    userType: 'Support             '
  },
  {
    active: true,
    email: 'anuj.gakhar@gmail.com',
    fullName: 'Anuj Gakhar',
    lastLogin: '1505204346',
    userId: 6064,
    userLevel: '4         ',
    userType: 'Support             '
  },
  {
    active: true,
    email: 'demogallery@artworkmanager.com',
    fullName: 'webmaster',
    lastLogin: '1505204346',
    userId: 9,
    userLevel: '4         ',
    userType: 'Webmaster           '
  },
  {
    active: true,
    email: 'ahershey79@yahoo.com',
    fullName: 'Andrew Hershey',
    lastLogin: '1505204346',
    userId: 6757,
    userLevel: '4         ',
    userType: 'SA                  '
  },
  {
    active: true,
    email: 'hello@managedartwork.com',
    fullName: 'Andrew Hershey',
    lastLogin: '1505204346',
    userId: 4975,
    userLevel: '4         ',
    userType: 'Support             '
  },
  {
    active: true,
    email: 'ellen@ellenmalloy.com',
    fullName: 'Ellen Malloy',
    lastLogin: '1505204346',
    userId: 6796,
    userLevel: '4         ',
    userType: 'Staff               '
  },
  {
    active: false,
    email: 'robin@galleryondivision.com',
    fullName: 'Robin Cairl',
    lastLogin: '1505204346',
    userId: 6795,
    userLevel: '3         ',
    userType: 'Gallery             '
  },
  {
    active: true,
    email: 'andrew@managedartwork.com',
    fullName: 'Andrew Hershey',
    lastLogin: '1505204346',
    userId: 6748,
    userLevel: '3         ',
    userType: 'Gallery             '
  },
  {
    active: true,
    email: 'grend500@gmail.com',
    fullName: 'Yemi',
    lastLogin: '1505204346',
    userId: 6761,
    userLevel: '4         ',
    userType: 'Support             '
  },
  {
    active: true,
    email: 'rumrunner1@gmail.com',
    fullName: 'Dan Sweet',
    lastLogin: '1505204346',
    userId: 5260,
    userLevel: '4         ',
    userType: 'Support             '
  },
  {
    active: true,
    email: 'karl@managedartwork.com',
    fullName: 'Karl Wiese',
    lastLogin: '1505204346',
    userId: 5455,
    userLevel: '4         ',
    userType: 'Support             '
  }
];
const pagination = {
  currentPage: 1,
  totalPages: 2,
  totalRecordCount: 11,
  recordsPerPage: "10",
}
const props = {
  list: [],
  appDispatch: jest.fn(),
  t: jest.fn(),
  pagination: {
    currentPage: 1,
    totalPages: 1,
    totalRecordCount: 0,
  },
};
describe('UsersGriddle', () => {
  it('Snapshot of without data', () => {
    const wrapper = shallow(<UsersGriddle {...props} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it('Snapshot with data', () => {
    const dProps = { ...props, ...{ list: userList } };
    const wrapper = shallow(<UsersGriddle {...dProps} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
