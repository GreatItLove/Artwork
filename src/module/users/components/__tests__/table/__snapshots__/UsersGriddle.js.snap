// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersGriddle Snapshot of without data 1`] = `
<div>
  <Griddle
    components={
      Object {
        "Filter": [Function],
        "NextButton": [Function],
        "PageDropdown": [Function],
        "PreviousButton": [Function],
        "SettingsWrapper": [Function],
        "Table": [Function],
      }
    }
    data={Array []}
    events={
      Object {
        "onFilter": [Function],
        "onGetPage": [Function],
        "onNext": [Function],
        "onPrevious": [Function],
      }
    }
    pageProperties={
      Object {
        "currentPage": 1,
        "pageSize": undefined,
        "recordCount": 0,
      }
    }
    styleConfig={
      Object {
        "classNames": Object {
          "Table": "table table-hover",
        },
      }
    }
  >
    <RowDefinition>
      <ColumnDefinition
        customComponent={[Function]}
        id="fullName"
      />
      <ColumnDefinition
        id="email"
      />
      <ColumnDefinition
        customComponent={[Function]}
        id="lastLogin"
      />
      <ColumnDefinition
        id="userType"
      />
      <ColumnDefinition
        id="userLevel"
      />
      <ColumnDefinition
        customComponent={[Function]}
        id="active"
      />
      <ColumnDefinition
        customComponent={[Function]}
        extraData={
          Object {
            "appDispatch": [Function],
          }
        }
        id="userId"
        title=" "
      />
    </RowDefinition>
  </Griddle>
  <div
    className="div-center show-per-page"
  >
    <text
      className="div-center users show"
    >
      Show
    </text>
     
    <CustomShowPerPage
      appDispatch={[Function]}
      fetchRequest={[Function]}
      list={Array []}
      pagination={
        Object {
          "currentPage": 1,
          "totalPages": 1,
          "totalRecordCount": 0,
        }
      }
      t={[Function]}
    />
    <text
      className="div-center users page"
    >
      per Page
    </text>
  </div>
  <Connect(Component) />
  <Connect(Component) />
  <Connect(AddUserDialog) />
  <Connect(EditUserDialog) />
  <Connect(ResetPasswordDialog) />
</div>
`;

exports[`UsersGriddle Snapshot with data 1`] = `
<div>
  <Griddle
    components={
      Object {
        "Filter": [Function],
        "NextButton": [Function],
        "PageDropdown": [Function],
        "PreviousButton": [Function],
        "SettingsWrapper": [Function],
        "Table": [Function],
      }
    }
    data={
      Array [
        Object {
          "active": true,
          "email": "demo@artworkmanager.com",
          "fullName": "Andrew Hershey",
          "lastLogin": "1505204346",
          "userId": 6733,
          "userLevel": "4         ",
          "userType": "Support             ",
        },
        Object {
          "active": true,
          "email": "anuj.gakhar@gmail.com",
          "fullName": "Anuj Gakhar",
          "lastLogin": "1505204346",
          "userId": 6064,
          "userLevel": "4         ",
          "userType": "Support             ",
        },
        Object {
          "active": true,
          "email": "demogallery@artworkmanager.com",
          "fullName": "webmaster",
          "lastLogin": "1505204346",
          "userId": 9,
          "userLevel": "4         ",
          "userType": "Webmaster           ",
        },
        Object {
          "active": true,
          "email": "ahershey79@yahoo.com",
          "fullName": "Andrew Hershey",
          "lastLogin": "1505204346",
          "userId": 6757,
          "userLevel": "4         ",
          "userType": "SA                  ",
        },
        Object {
          "active": true,
          "email": "hello@managedartwork.com",
          "fullName": "Andrew Hershey",
          "lastLogin": "1505204346",
          "userId": 4975,
          "userLevel": "4         ",
          "userType": "Support             ",
        },
        Object {
          "active": true,
          "email": "ellen@ellenmalloy.com",
          "fullName": "Ellen Malloy",
          "lastLogin": "1505204346",
          "userId": 6796,
          "userLevel": "4         ",
          "userType": "Staff               ",
        },
        Object {
          "active": false,
          "email": "robin@galleryondivision.com",
          "fullName": "Robin Cairl",
          "lastLogin": "1505204346",
          "userId": 6795,
          "userLevel": "3         ",
          "userType": "Gallery             ",
        },
        Object {
          "active": true,
          "email": "andrew@managedartwork.com",
          "fullName": "Andrew Hershey",
          "lastLogin": "1505204346",
          "userId": 6748,
          "userLevel": "3         ",
          "userType": "Gallery             ",
        },
        Object {
          "active": true,
          "email": "grend500@gmail.com",
          "fullName": "Yemi",
          "lastLogin": "1505204346",
          "userId": 6761,
          "userLevel": "4         ",
          "userType": "Support             ",
        },
        Object {
          "active": true,
          "email": "rumrunner1@gmail.com",
          "fullName": "Dan Sweet",
          "lastLogin": "1505204346",
          "userId": 5260,
          "userLevel": "4         ",
          "userType": "Support             ",
        },
        Object {
          "active": true,
          "email": "karl@managedartwork.com",
          "fullName": "Karl Wiese",
          "lastLogin": "1505204346",
          "userId": 5455,
          "userLevel": "4         ",
          "userType": "Support             ",
        },
      ]
    }
    events={
      Object {
        "onFilter": [Function],
        "onGetPage": [Function],
        "onNext": [Function],
        "onPrevious": [Function],
      }
    }
    pageProperties={
      Object {
        "currentPage": 1,
        "pageSize": undefined,
        "recordCount": 0,
      }
    }
    styleConfig={
      Object {
        "classNames": Object {
          "Table": "table table-hover",
        },
      }
    }
  >
    <RowDefinition>
      <ColumnDefinition
        customComponent={[Function]}
        id="fullName"
      />
      <ColumnDefinition
        id="email"
      />
      <ColumnDefinition
        customComponent={[Function]}
        id="lastLogin"
      />
      <ColumnDefinition
        id="userType"
      />
      <ColumnDefinition
        id="userLevel"
      />
      <ColumnDefinition
        customComponent={[Function]}
        id="active"
      />
      <ColumnDefinition
        customComponent={[Function]}
        extraData={
          Object {
            "appDispatch": [Function],
          }
        }
        id="userId"
        title=" "
      />
    </RowDefinition>
  </Griddle>
  <div
    className="div-center show-per-page"
  >
    <text
      className="div-center users show"
    >
      Show
    </text>
     
    <CustomShowPerPage
      appDispatch={[Function]}
      fetchRequest={[Function]}
      list={
        Array [
          Object {
            "active": true,
            "email": "demo@artworkmanager.com",
            "fullName": "Andrew Hershey",
            "lastLogin": "1505204346",
            "userId": 6733,
            "userLevel": "4         ",
            "userType": "Support             ",
          },
          Object {
            "active": true,
            "email": "anuj.gakhar@gmail.com",
            "fullName": "Anuj Gakhar",
            "lastLogin": "1505204346",
            "userId": 6064,
            "userLevel": "4         ",
            "userType": "Support             ",
          },
          Object {
            "active": true,
            "email": "demogallery@artworkmanager.com",
            "fullName": "webmaster",
            "lastLogin": "1505204346",
            "userId": 9,
            "userLevel": "4         ",
            "userType": "Webmaster           ",
          },
          Object {
            "active": true,
            "email": "ahershey79@yahoo.com",
            "fullName": "Andrew Hershey",
            "lastLogin": "1505204346",
            "userId": 6757,
            "userLevel": "4         ",
            "userType": "SA                  ",
          },
          Object {
            "active": true,
            "email": "hello@managedartwork.com",
            "fullName": "Andrew Hershey",
            "lastLogin": "1505204346",
            "userId": 4975,
            "userLevel": "4         ",
            "userType": "Support             ",
          },
          Object {
            "active": true,
            "email": "ellen@ellenmalloy.com",
            "fullName": "Ellen Malloy",
            "lastLogin": "1505204346",
            "userId": 6796,
            "userLevel": "4         ",
            "userType": "Staff               ",
          },
          Object {
            "active": false,
            "email": "robin@galleryondivision.com",
            "fullName": "Robin Cairl",
            "lastLogin": "1505204346",
            "userId": 6795,
            "userLevel": "3         ",
            "userType": "Gallery             ",
          },
          Object {
            "active": true,
            "email": "andrew@managedartwork.com",
            "fullName": "Andrew Hershey",
            "lastLogin": "1505204346",
            "userId": 6748,
            "userLevel": "3         ",
            "userType": "Gallery             ",
          },
          Object {
            "active": true,
            "email": "grend500@gmail.com",
            "fullName": "Yemi",
            "lastLogin": "1505204346",
            "userId": 6761,
            "userLevel": "4         ",
            "userType": "Support             ",
          },
          Object {
            "active": true,
            "email": "rumrunner1@gmail.com",
            "fullName": "Dan Sweet",
            "lastLogin": "1505204346",
            "userId": 5260,
            "userLevel": "4         ",
            "userType": "Support             ",
          },
          Object {
            "active": true,
            "email": "karl@managedartwork.com",
            "fullName": "Karl Wiese",
            "lastLogin": "1505204346",
            "userId": 5455,
            "userLevel": "4         ",
            "userType": "Support             ",
          },
        ]
      }
      pagination={
        Object {
          "currentPage": 1,
          "totalPages": 1,
          "totalRecordCount": 0,
        }
      }
      t={[Function]}
    />
    <text
      className="div-center users page"
    >
      per Page
    </text>
  </div>
  <Connect(Component) />
  <Connect(Component) />
  <Connect(AddUserDialog) />
  <Connect(EditUserDialog) />
  <Connect(ResetPasswordDialog) />
</div>
`;
